### PR TITLE
chore(root): add automated cherry-picking to v3/develop

### PR DESCRIPTION
**Pull request by evenstensberg**
## Summary of the changes
Automates cherry-picking (https://github.com/marketplace/actions/github-cherry-pick-action). To convert this from draft to ready, enable the action and merge this PR.

## Related issue
#2669

### General 

- [x] Changes to docs package checked and committed.
- [x] All acceptance criteria reviewed and met. 

### Testing

- [x] Relevant unit tests and visual regression tests added. 
- [x] Visual testing against Figma component specification completed. 
- [x] Playground stories in React Storybook up to date, with any prop changes and additions addressed. 
- [x] Compare performance of modified components against develop using Performance addon in React Storybook.

### Accessibility 

- [x] Accessibility Insights FastPass performed.
- [x] A11y unit test added and yields no issues.
- [x] A11y plug-in on Storybook yields no issues. 
- [x] Manual screen reader testing performed using NVDA and VoiceOver. 
- [x] Manual keyboard testing for keyboard controls and logical focus order. 
- [x] Correct roles used and ARIA attributes used correctly where required. 
- [x] Logical heading structure is maintained, and the HTML elements used for headings can be changed to fit within the wider page structure. 

### Resize/zoom behaviour 

- [x] Page can be zoomed to 400% with no loss of content. 
- [x] Screen magnifier used with no issues. 
- [x] Text resized to 200% with no loss of content.
- [x] Text spacing increased as per the [WCAG 1.4.12 success criterion](https://www.w3.org/TR/WCAG21/#text-spacing) with no loss of content.

### System modes

- [x] Browser setting 'prefers reduced motion' tested. No animations or motion visible whilst this setting is on. 
- [x] Windows High Contrast mode tested with no loss of content.
- [x] System light and dark mode tested with no loss of content.
- [x] Browser support tested (Chrome, Safari, Firefox and Edge). 

### Testing content extremes

- [x] Min/max content examples tested with no loss of content or overflow. 
- [x] All prop combinations work without issue. 
- [x] Tested for FOUC (Flash of Unstyled Content) in both SSR (Server-Side Rendering) and SSG (Static Site Generation) settings.
- [x] Controlled and uncontrolled input components tested.
- [x] Props/slots can be updated after initial render.

### CODE REVIEW

**Comment by evenstensberg** [comment]
CC @GCHQ-Developer-530

**Comment by GCHQ-Developer-530** [comment]
Hi @evenstensberg thank you so much for doing this so quickly! It's looking good, just one change I think that's needed is that we'd like the work to be cherry picked onto a new branch (with a base branch of v3.0.0/develop), and then have the action open a pull request from that new branch onto v3.0.0/develop. We've had a few issues with conflicts when cherry picking due to prop name changes, so it would be easier to fix a pull request than fix v3.0.0/develop 👍

**Comment by evenstensberg** [comment]
@GCHQ-Developer-530 If I understood, the logic is this:

1. commit merged in develop
2. The action then cherry-picks that PR it was merged from, onto a new PR to the v3.0.0/develop branch


I think this is the intended behaviour, no?

**Comment by GCHQ-Developer-530** [comment]
Sorry yes that's correct, I misunderstood the action and thought that it was cherry picking directly onto v3.0.0/develop. Thank you for explaining!

**Review by GCHQ-Developer-530 — APPROVED** [review]

File: .github/cherry-pick.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  pull_request:
+    branches:
+      - develop
+    types: ["closed"]
+
+jobs:
+  cherry_pick_release_v3_0:
+    runs-on: ubuntu-latest
+    name: Cherry pick into v3.0.0/develop
+    if: ${{ contains(github.event.pull_request.labels.*.name, 'v3.0.0/develop') && github.event.pull_request.merged == true }}
```
Comment (ad9242) [inline .github/cherry-pick.yml:11]: should this be 'develop' rather than 'v3.0.0/develop'? as the change would have been merged into `develop` branch

Comment (ad9242) [reply .github/cherry-pick.yml:11]: in fact, can the check just be:
```suggestion
    if: ${{ github.event.pull_request.merged == true }}
```
as this workflow should only be running against the `develop` branch, shouldn't it?

**Review by ad9242 — CHANGES_REQUESTED** [review]

File: .github/cherry-pick.yml (hunk 1 above)
Comment (evenstensberg) [reply .github/cherry-pick.yml:11]: Yes, that is correct.

**Review by GCHQ-Developer-530 — APPROVED** [review]

**Review by ad9242 — APPROVED** [review]

**Pull request merged**